plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/JdbcMetadataFactory.java

```diff
@@ -39,6 +41,6 @@ public JdbcMetadata create()
     {
         // Session stays the same per transaction, therefore session properties don't need to
         // be a part of cache keys in CachingJdbcClient.
-        return new JdbcMetadata(new CachingJdbcClient(jdbcClient, Set.of(), new Duration(1, DAYS), true), allowDropTable);
+        return new JdbcMetadata(new CachingJdbcClient(jdbcClient, Set.of(), identityMapping, new Duration(1, DAYS), true), allowDropTable);
```

Comment:
identity stays the same per transaction, therefore identity mapping doesn't need to be part of the cache key

we should use SingletonJdbcIdentityCacheMapping here